include/VerificationValidation.h

```diff
@@ -8,40 +8,106 @@
 #include <list>
 #include <map>
 #include <QString>
+#include "ObjectTreeWidget.h"
 
 namespace VerificationValidation {
     class Arg {
     public:
+        enum Type {
+            Static, // argument is not variable (e.g.: "-t")
+            Dynamic, // argument is variable (e.g.: "-t3mm,3mm" or "-t5mm,5mm")
+            ObjectNone, // only used to determine when a command has no object
+            ObjectName, // argument is the objectName
+            ObjectPath // argument is the objectPath
+        };
+
         QString argument;
-        bool isVariable;
+        int argIdx;
         QString defaultValue;
+        Type type;
 
-        Arg(QString argument, bool isVariable, QString defaultValue){
+        Arg(QString argument, QString defaultValue = NULL, Type type = Static){
             this->argument = argument;
-            this->isVariable = isVariable;
             this->defaultValue = defaultValue;
-        }
 
-        void updateValue (QString input){
-            defaultValue = input;
+            if (defaultValue != NULL) this->type = Dynamic;
+            else this->type = type;
         }
+
+        bool operator<(const Arg& rhs) { return argument < rhs.argument; }
     };
 
     class Test {
     public:
         QString testName;
-        QString testCommand;
-        QString suiteName;
+        QStringList suiteNames;
         QString category;
-        bool hasVariable;
         std::vector<Arg> ArgList;
 
-        QString getCmdWithArgs() const {
-            QString cmd = testCommand;
+        Test(const QString& testName, const QStringList& suiteNames, const std::vector<Arg>& ArgList) :
+        testName(testName), suiteNames(suiteNames), ArgList(ArgList),
+        category((ArgList.size()) ? ArgList[0].argument : "NULL")
+        {
+            for (int i = 0; i < this->ArgList.size(); i++)
+                this->ArgList[i].argIdx = i;
+        }
+
+        bool operator==(const Test& rhs) {
+            if (ArgList.size() != rhs.ArgList.size()) return false;
+
+            std::vector<Arg> lhsArgList(ArgList);
+            std::vector<Arg> rhsArgList(rhs.ArgList);
+            std::sort(lhsArgList.begin(), lhsArgList.end());
+            std::sort(rhsArgList.begin(), rhsArgList.end());
+
+            for (int i = 0; i < lhsArgList.size(); i++) {
+                if (lhsArgList[i].type == rhsArgList[i].type && (lhsArgList[i].type == Arg::Type::ObjectName || lhsArgList[i].type == Arg::Type::ObjectNone || lhsArgList[i].type == Arg::Type::ObjectPath))
+                    continue;
+                if (lhsArgList[i].argument != rhsArgList[i].argument)
```

Comment:
Need to also compare defaultValue -- noticed an error where No Overlap tests were being run as No Null Region tests because the arguments are identical but the default values are not.

Reply:
great catch, changes pushed